src/zscale.jl

```diff
@@ -0,0 +1,248 @@
+MAX_REJECT = 0.5
+MIN_NPIXELS = 5
+GOOD_PIXEL = 0
+BAD_PIXEL = 1
+KREJ = 2.5
+MAX_ITERATIONS = 5
+
+
+function flatten(image)
+
+    # colapses into one dimension (Float64) a 2D array in row-major (C-style flattening)
+
+    rows = size(image)[1]
+    cols = size(image)[2]
+    n_pxls = rows * cols
+
+    result = Array{Float64}(undef, n_pxls)
+    
+    k = 1
+    for i in 1:rows, j in 1:cols
+        result[k] = image[i, j]
+        k += 1
+    end
+
+    return result
+end
+
+
+function zsc_sample(image, maxpix)
+
+    # Figure out which pixels to use for the zscale algorithm
+    # Returns the 1-d array samples
+    # Sample in a square grid, and return the first maxpix in the sample
+
+    nc = size(image)[1]
+    nl = size(image)[2]
+
+    stride = maximum([1.0, sqrt((nc - 1) * (nl - 1) / maxpix)])
+    stride = Int128(round(stride))
+
+    samples = flatten(image[1:stride:end, 1:stride:end])
+    
+    return samples[1:maxpix]
+end
+
+
+function zsc_compute_sigma(flat, badpix)
```

Comment:
Also here (and everywhere) a docstring would be useful.  Please also explain the meaning of the `mean` or `sigma` equal to `nothing` (which I'm not sure is a good idea)